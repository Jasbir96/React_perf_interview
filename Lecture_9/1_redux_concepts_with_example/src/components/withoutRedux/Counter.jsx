import { useState } from 'react';
const Counter = () => {
    const [count, setCount] = useState(0);

    const handleIncrement = () => {
       setCount(count+1);
    }

    const handleDecrement = () => {
        setCount(count-1);
    }

    return (
        <>
            <h2>Counter without redux</h2>
            <button onClick={handleIncrement}>+</button>
            <p>count: {count}</p>
            <button onClick={handleDecrement}>-</button>
        </>
    )
}


export default Counter;