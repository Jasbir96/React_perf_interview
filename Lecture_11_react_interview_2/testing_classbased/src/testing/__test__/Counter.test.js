import { render, screen, fireEvent } from '@testing-library/react';
import Counter from '../Counter';

describe("counter test cases", () => {
    test("snapshot for counter", () => {
        const { asFragment } = render(<Counter></Counter>);
        expect(asFragment()).toMatchSnapshot();
    })
    test("inital state check", () => {
        //1. render
        render(<Counter></Counter>)
        // select
        const countText = screen.getByText("Counter");
        const CountValue = screen.getByText("0");
        const plusText = screen.getByText("+");
        const minusText = screen.getByText("-");

        // 3. compare
        expect(countText).toBeInTheDocument();
        expect(CountValue).toBeInTheDocument();
        expect(plusText).toBeInTheDocument();
        expect(minusText).toBeInTheDocument();
    })

    test("inc by 1", () => {
        // 1. it is in the UI 
        render(<Counter></Counter>)
        const plusText = screen.getByText("+");
        fireEvent.click(plusText);
        const isOnePresent = screen.getByText("1");
        expect(isOnePresent).toBeInTheDocument();
    })
    test("dec by 1", () => {

        render(<Counter></Counter>);

        const minusText = screen.getByText("-");

        fireEvent.click(minusText);
        fireEvent.click(minusText);

        const isOnePresent = screen.getByText("-2");
        expect(isOnePresent).toBeInTheDocument();
    })
})
