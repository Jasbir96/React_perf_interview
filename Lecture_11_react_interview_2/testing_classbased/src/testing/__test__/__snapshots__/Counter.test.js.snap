// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter test cases snapshot for counter 1`] = `
<DocumentFragment>
  <h1>
    Counter
  </h1>
  <h2>
    0
  </h2>
  <div>
    <button>
      +
    </button>
    <button>
      -
    </button>
  </div>
</DocumentFragment>
`;
